qiita_db/parameters.py

```diff
@@ -172,6 +172,27 @@ def to_str(self):
 
         return " ".join(result)
 
+    @classmethod
+    def get_all_parameters(self):
+        r""" Generates a dictionary of all the available parameters
+
+        Returns
+        -------
+        dict
+            A dictionary of column_name: value
+        """
+        conn_handler = SQLConnectionHandler()
+
+        columns = [r[0] for r in conn_handler.execute_fetchall(
+            "SELECT column_name FROM information_schema.columns WHERE "
```

Comment:
We should update all our code to depend on the current release version of
QIIME as it should not be any further release on that 1.X branch.

2015-02-17 12:15 GMT-08:00 Daniel McDonald <notifications@github.com>:

> In qiita_db/parameters.py
> <https://github.com/biocore/qiita/pull/859#discussion_r24849687>:
>
> > @@ -172,6 +172,27 @@ def to_str(self):
> >
> >          return " ".join(result)
> >
> > +    @classmethod
> > +    def get_all_parameters(self):
> > +        r""" Generates a dictionary of all the available parameters
> > +
> > +        Returns
> > +        -------
> > +        dict
> > +            A dictionary of column_name: value
> > +        """
> > +        conn_handler = SQLConnectionHandler()
> > +
> > +        columns = [r[0] for r in conn_handler.execute_fetchall(
> > +            "SELECT column_name FROM information_schema.columns WHERE "
>
> It kinda depends on which commit... for instance, the dash/underscore
> issue that @adamrp <https://github.com/adamrp> referred to was not in
> place when I put in the fastq demux processing which was after the
> Flagstaff sprint. If qiime on qiita.colorado.edu has been updated since,
> then it is possible there is an inconsistency with the params table, that,
> or I may have omitted some of the ambig options at the time
> … <#14b992e6863b8a72_>
> On Tue, Feb 17, 2015 at 1:11 PM, Antonio Gonzalez <
> notifications@github.com> wrote: In qiita_db/parameters.py <
> https://github.com/biocore/qiita/pull/859#discussion_r24849395>: > @@
> -172,6 +172,27 @@ def to_str(self): > > return " ".join(result) > > +
> @classmethod <https://github.com/classmethod> > + def
> get_all_parameters(self): > + r""" Generates a dictionary of all the
> available parameters > + > + Returns > + ------- > + dict > + A dictionary
> of column_name: value > + """ > + conn_handler = SQLConnectionHandler() > +
> > + columns = [r[0] for r in conn_handler.execute_fetchall( > + "SELECT
> column_name FROM information_schema.columns WHERE " 1.8.0-dev so really
> 1.9.0. — Reply to this email directly or view it on GitHub <
> https://github.com/biocore/qiita/pull/859/files#r24849395>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/qiita/pull/859/files#r24849687>.
>



-- 
Jose Navas